### PR TITLE
Changing most references in our codebase from master to main

### PR DESCRIPTION
**Pull request by Maxrimus**
Preparing for master to main switch by changing our references in our repo from master to main
Signed-off-by: Joe Tursi <joe.p.tursi@gmail.com>

### CODE REVIEW

File: doc/rst/developer/bestPractices/#git.rst# [hunk 1]
```diff
@@ -0,0 +1,477 @@
+Git tips for Chapel developers
```
Comment (bradcray) [inline doc/rst/developer/bestPractices/#git.rst#:1]: This file looks like it was git added by accident?

File: doc/rst/developer/bestPractices/.#git.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+joe@C02YR32ZLVCJ.898
```
Comment (bradcray) [inline doc/rst/developer/bestPractices/.#git.rst:1]: This one too?

File: doc/rst/developer/chips/15.rst [hunk 3]
```diff
@@ -232,7 +232,7 @@ related issues (mentioned later on).
 
 Another reason why we decided not to move with our approach based on
 disabling dead code elimination was to avoid having a large
-departure from how the same file would appear on master. Which would
+departure from how the same file would appear on main. Which would
```
Comment (bradcray) [inline doc/rst/developer/chips/15.rst:235]: I wonder whether "on main" is somewhat ambiguous without more context.  Maybe:

* "on 'main'" to indicate that 'main' isn't just the English word, but something special?
* "on the 'main' branch" (or without quotes now that it's not serving as a special identifier as much)
* "on the 'main' development branch" (ditto)

File: doc/rst/developer/chips/inactive/23.rst [hunk 4]
```diff
@@ -760,7 +760,7 @@ the convention of placing complete() as the final statement in the
 body; this will simplify a process of finding these cases and
 removing the unnecessary complete().
 
-These updates can be applied to master in any number of stages
+These updates can be applied to main in any number of stages
```
Comment (bradcray) [inline doc/rst/developer/chips/inactive/23.rst:763]: Similar comments as the previous, though here using a quoted 'main' seems sufficiently clear to me.

File: test/classes/delete-free/lifetimes/array-borrows-owns-issue.chpl [hunk 5]
```diff
@@ -1,4 +1,4 @@
-// This example doesn't compile on master now even though
+// This example doesn't compile on main now even though
```
Comment (bradcray) [inline test/classes/delete-free/lifetimes/array-borrows-owns-issue.chpl:1]: quotes?

**Comment by bradcray** [comment]
* Should we change the URLs in CHANGES.md to refer to `blob/main` rather than `blob/master`?  (I think so)
* It looks to me like git.rst still needs updating; I'm guessing that this relates to the #git.rst#-style files that look like they were accidentally committed
* `test/classes/initializers/generics/declarations/*.future` looks like it still has a bunch of references to `master` similar to a class of cases you already changed?

This can be separate from this PR, but I think should be tackled as part of this same effort:
* To complete this change, we'll ultimately also want to update the chapel-lang.org to use `docs/main` rather than (or in addition to for some time) `docs/master`
  * I would guess that doc/Makefile is related to this
  * I'll also likely need to mess with some symbolic links (and maybe scripts?) in `chapel-www`

**Comment by Maxrimus** [comment]
> * Should we change the URLs in CHANGES.md to refer to `blob/main` rather than `blob/master`?  (I think so)
My thought to not changing them was that the branch WAS master at the time of that release, whereas it WILL be main going forward, so we should definitely change it in future reviews. Unclear to me whether that distinction matters, if you think I should change them I will.
> * It looks to me like git.rst still needs updating; I'm guessing that this relates to the #git.rst#-style files that look like they were accidentally committed
Yea for some reason those two were committed but the regular git.rst wasn't, going to fix that in the next push.
> * `test/classes/initializers/generics/declarations/*.future` looks like it still has a bunch of references to `master` similar to a class of cases you already changed?
I did make changes to those .future files, unless I missed some?

**Comment by bradcray** [comment]
> My thought to not changing them was that the branch WAS master at the time of that release, whereas it WILL be main going forward, so we should definitely change it in future reviews. Unclear to me whether that distinction matters, if you think I should change them I will.

That's definitely an interesting tension, and one that I wrestle with in other situations w.r.t. the CHANGES file.  If it were just a text file, I'd tend to agree with what you're saying, but since viewing it on GitHub causes the URLs to be treated as hyperlinks—https://github.com/chapel-lang/chapel/blob/master/CHANGES.md—I think it'd be shame to have a bunch of broken hyperlinks if they could just as easily point to something that works (if GitHub will forward `master/` to `main/` in such URLs for the long-term, then I don't feel as worried about this... for some reason I thought that they might be doing something along those lines to smooth over these transitions, but am not confident and haven't looked into it).

> unless I missed some?

On my local copy of your branch, I'm seeing:

```
$ git grep master | grep future
classes/initializers/generics/declarations/concrete-inits.future:is on master.  Make it a future so that Lydia can stay in step with master
classes/initializers/generics/declarations/default_value.future:is on master.  Make it a future so that Lydia can stay in step with master
classes/initializers/generics/declarations/no-arg-param.future:is on master.  Make it a future so that Lydia can stay in step with master
classes/initializers/generics/declarations/no-arg-type.future:is on master.  Make it a future so that Lydia can stay in step with master
```

so unless I've done something wrong (and I wouldn't put it past me), it looked like you had.

**Review by bradcray — APPROVED** [review]

**Comment by bradcray** [comment]
This looks good to me.  I don't feel 100% confident we haven't missed something, but I think we got the vast majority of it.

Will things need to change in our testing scripts to keep things working (e.g., in the 'code' and 'ci' configurations)?

**Pull request merged**